src/AvPortalClient.php

```
@@ -68,13 +91,13 @@ public function __construct(ClientInterface $httpClient, ConfigFactoryInterface
    * {@inheritdoc}
    */
   public function query(array $options = []): ?array {
-    $options += [
-      'fl' => 'type,ref,doc_ref,titles_json,duration,shootstartdate,media_json,mediaorder_json,summary_json,languages',
-      'hasMedia' => 1,
-      'wt' => 'json',
-      'index' => 1,
-      'pagesize' => 15,
-    ];
+    try {
+      $options = $this->buildOptions($options);
+    }
+    catch (\OutOfRangeException $exception) {
+      $this->logger->error($exception->getMessage());
+      return $this->resourcesFromResponse([]);
```

Passing wrong parameters shouldn't be intercepted and logged, it should throw an exception instead.
See [the Symfony options resolver component](https://symfony.com/doc/current/components/options_resolver.html). In case a wrong option is passed, [an exception is thrown](https://symfony.com/doc/current/components/options_resolver.html#value-validation).

Logging should/could be done when a failure to do the request occurs (see the @todo part) as it's something not fault of the user of the client.

Let's revert this part with the logger back.